_gsocblogs/2022/blog_ATHENA_UjwalKundur.md

```diff
@@ -57,17 +67,72 @@ Now I could move onto the actual C++ code!
+
+However, the final results were pretty disappointing. SYCL code performed worse than "normal" sequential code at varying workloads. Why? No GPU support - During this time I realized that the Intel provided OneAPI containers that oneapi_jug_xl was based on did not provide **CUDA** as a backend as it was still in experimental stage. This was another blow to an already difficult project. This meant we had to re-evaluate the entire SYCL setup and it was agreed upon that this was out of scope of the current project.
+
+Since we had no access to AMD GPUs - which support OpenCL by default (the GPGPU scene is dominated by NVidia and CUDA), I had to settle for a performance comparison of SYCL and Non-SYCL code on the same AMD EPYC 32-core CPU used in Compute Canada's HPC systems. Hence, the comparison results you see below are sub-par and show poor performance on SYCL's part. However, I believe that given proper hardware, software stack and tuning, many algorithms in Juggler will greatly benefit from SYCL support.
+
+![Wall-Time Cmpr](https://github.com/Ajax-Light/GSoC-cernhsf-final/raw/master/reports/WallTime-compare.png)
+
+The SYCLized code was merged into master, Merge Request [here](https://eicweb.phy.anl.gov/EIC/juggler/-/merge_requests/469).
+
+## Conclusion
+
+Few things which remain to be worked on are:
+
+* Smaller image size for `oneapi_jug_xl`
+* Adding `CUDA` support to the oneapi containers
+* Further optimizations and tuning for the SYCLized `IslandCluster` Algorithm
+
+While the project has been *way*, *way* harder to work on than I had initially anticipated, it was incredibly rewarding and I'm grateful to have had the opportunity to contribute to such an exotic project and work with amazing people in a completely different discipline from mine.
+
+I cannot thank my mentors, Dr. Wouter Deconinck and Sylvester Joosten enough for their help and for being so patient with me even when I bombard them with questions :)
+
+I hope to continue contributing to Juggler and the EIC Software after GSoC as well and am looking forward to see the software in action in the near future.
 
-It has been a great learning experience and I'm really grateful to have awesome mentors and a cool community which I can be a part of. So Thanks CERN-HSF!!
+If you've reached the bottom of this page, Thank You!
 
-If you've reached the bottom of this page, Thank You! See you in a month,
 Cheers,
 Ujwal.
 
 ---
 
-Reach out to me at: ujwal.kundur@gmail.com \|\| [GitHub](https://github.com/Ajax-Light)
+Reach out to me at: ujwal.kundur@gmail.com \|\| [GitHub](https://github.com/Ajax-Light) \|\| [LinkedIn](https://www.linkedin.com/in/ujwal-kundur/)
```

Comment:
@Ajax-Light your LinkedIn profile link cannot be viewed, probably because you did not make it fully public. Please fix this.

Reply:
Oh was the profile visibility the issue? I've changed it from LinkedIn members to public to help the bot. Should I push something to trigger the workflow again or is it unnecessary?

Reply:
I can remove the link if this doesn't work

Reply:
I have rerun the check, it still fails, and indeed if I click the link it gets me to the LinkedIn login page.

Reply:
@Ajax-Light I think you need to provide the link as: https://in.linkedin.com/in/ujwal-kundur

Reply:
It still fails, I'll remove it.